packages/mongodb-log-writer/src/mongo-log-manager.ts

```diff
@@ -39,42 +40,63 @@ export class MongoLogManager {
     if (options.prefix) {
       if (!/^[a-z0-9_]+$/i.test(options.prefix)) {
         throw new Error(
-          'Prefix must only contain letters, numbers, and underscores'
+          'Prefix must only contain letters, numbers, and underscores',
         );
       }
     }
     this._options = options;
   }
 
-  private async deleteFile(path: string): Promise<void> {
+  private async deleteFile(filePath: string): Promise<void> {
     try {
-      await fs.unlink(path);
+      await fs.unlink(filePath);
       // eslint-disable-next-line @typescript-eslint/no-explicit-any
     } catch (err: any) {
       if (err?.code !== 'ENOENT') {
-        this._options.onerror(err as Error, path);
+        this._options.onerror(err as Error, filePath);
       }
     }
+
+    let basePath = path.dirname(filePath);
+    while (basePath !== this._options.directory) {
```

Comment:
What if `this._options.directory` was specified as `/foo/bar/`? Repeatedly calling `path.dirname()` on `/foo/bar/baz` will never reach that filename.

I'd imagine that `path.normalize()` together with some sort of prefix check would work here, but I'm not sure if that's the most reliable way to do this either